ci/scripts/coverage.sh

```diff
@@ -137,6 +146,36 @@ function main() {
     show_header "R package coverage summary"
     Rscript -e 'library(covr); print(readRDS("r_coverage.rds"))'
     popd
+
+    # Build + test Python package with cython/gcc coverage options
+    show_header "Build + test Python package"
+    pushd "${SANDBOX_DIR}"
+    TARGET_NANOARROW_PYTHON_DIR="${TARGET_NANOARROW_DIR}/python"
+
+    pushd "${TARGET_NANOARROW_PYTHON_DIR}"
+    python -m pip install -e .
+    NANOARROW_COVERAGE=1 python setup.py build_ext --inplace
```

Comment:
```suggestion
    NANOARROW_COVERAGE=1 python -m pip install -e .
```

That doesn't work? (`pip install -e .` should essentially under the hood already be calling `python setup.py build_ext --inplace`, but then through official APIs)